### ISSUE TITLE
[BUG] filePattern parameter finds zero files when repository path contains substring matching exclude patterns

### ISSUE THREAD

**Issue by JonahSussman** [konveyor/analyzer-lsp#1083]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

Latest main branch (commit: fe83a80)

### Priority

Major

### Current Behavior

When using `builtin.filecontent` with the `filePattern` parameter, the analyzer sometimes finds zero files even when matching files exist in the repository.

This occurs when the repository path contains substrings that match default exclude directory patterns. For example, running the analyzer somewhere in a directory prefixed with `/home/user/Projects/github.com/...` and in a directory without `.git` can trigger the bug because `.git` (in the default excludes) gets treated as a regex and matches `/git` in the path.

### Expected Behavior

The `filePattern` parameter should find all files matching the pattern, regardless of the repository's absolute path. Exclude patterns like `.git` should only exclude literal `.git` directories, not paths containing the substring `/git`.

### How Reproducible

Always (Default)

### Steps To Reproduce

Run the following bash script:

```bash
#!/bin/bash
set -e

# Configuration
BRANCH="main"
REPO_DIR="path/to/repo"
TEST_DIR=$(mktemp -d -t analyzer-test-XXXXXX)

echo "=========================================="
echo "REPO_DIR: $REPO_DIR"
echo "BRANCH:   $BRANCH"
echo "TEST_DIR: $TEST_DIR"
echo "=========================================="
echo

# Navigate to repo and checkout branch
cd "$REPO_DIR"
echo "[1/6] Checking out branch: $BRANCH"
git checkout "$BRANCH" 2>&1 # | grep -v "^Already on"

# Build analyzer
echo "[2/6] Building analyzer..."
go build -o "$TEST_DIR/analyzer" ./cmd/analyzer 2>&1 | tail -5

# Create test directory with 'git' in name (triggers bug)
echo "[3/6] Creating test files..."
mkdir -p "$TEST_DIR/test-dir-with-git-in-name"

cat > "$TEST_DIR/test-dir-with-git-in-name/test.jsx" << 'JSXEOF'
import { Page, Masthead } from '@patternfly/react-core';

function MyComponent() {
  return (
    <Page header={<Masthead />} />
  );
}
JSXEOF

# Create test rule
cat > "$TEST_DIR/test-rule.yaml" << 'YAMLEOF'
---
- ruleID: test-filecontent-pattern
  labels:
  - konveyor-analysis
  description: Test filePattern parameter
  effort: 3
  category: potential
  when:
    builtin.filecontent:
      pattern: '@patternfly'
      filePattern: '\.([jt])sx?$'
  message: "Found pattern"
YAMLEOF

# Create provider settings (must use initConfig array format)
cat > "$TEST_DIR/provider-settings.json" << JSONEOF
[
  {
    "name": "builtin",
    "initConfig": [
      {
        "location": "$TEST_DIR/test-dir-with-git-in-name"
      }
    ]
  }
]
JSONEOF

# Run analyzer
echo "[4/6] Running analyzer..."
cd "$TEST_DIR"
./analyzer --label-selector konveyor-analysis \
  --rules test-rule.yaml \
  --provider-settings provider-settings.json \
  --output-file output.yaml 2>&1 | grep -E "error|processing rule|rule returned" || true

# Check results
echo
echo "[5/6] Results:"
echo "----------------------------------------"
cat output.yaml
echo "----------------------------------------"
echo

# Determine if bug exists
if grep -q "unmatched" output.yaml; then
    echo "[6/6] x Rule did not match (no files found)"
    EXIT_CODE=1
else
    echo "[6/6] o Rule matched successfully"
    EXIT_CODE=0
fi

echo
echo "Test directory: $TEST_DIR"
echo "To inspect: cd $TEST_DIR"
echo "To cleanup: rm -rf $TEST_DIR"

exit $EXIT_CODE
```

Actual result:
```yaml
- name: konveyor-analysis
  unmatched:
  - test-filecontent-pattern
```

Expected result:
```yaml
- name: konveyor-analysis
  violations:
    test-filecontent-pattern:
      description: Test filePattern parameter
      category: potential
      labels:
      - konveyor-analysis
      incidents:
      - uri: file:///tmp/analyzer-test-kuUZQV/test-dir-with-git-in-name/test.jsx
        message: Found pattern
        codeSnip: " 1  import { Page, Masthead } from '@patternfly/react-core';\n 2  \n 3  function MyComponent() {\n 4    return (\n 5      <Page header={<Masthead />} />\n 6    );\n 7  }\n"
        lineNumber: 1
        variables:
          matchingText: '@patternfly'
      effort: 3
```

### Environment

```markdown
N/A
```
